### DISCUSSION TITLE
Channel Image: ¿Cómo puedo cambiar la imagen de un canal?

### DISCUSSION THREAD

**Discussion by ValeMP** [Cotalker/documentation#67] [2. Questions and Answers] [answered]
Hola!
Quería poder cambiar la imagen de un canal para poder diferenciar distintos canales. 
@guille me menciono que hay algunas imagenes ya guardadas que pueden ser usadas, pero no sé donde podría cachar cuales son. Me podrían ayudar en decirme, como poder verlas? 

Solo necesito una que aparezca un usuario y otra que aparezca una maleta. Entonces, si puedo ver las imagenes, podría seleccionar alguna para mostrar. 

Gracias!

**Comment by acedward** [comment]
Existen URLs para la generación dinámica de imágenes para canales y avatares que siguen este formato 

![image](https://user-images.githubusercontent.com/3951895/114418872-eda40280-9b80-11eb-9d7a-421a104a8d39.png)

https://www.cotalker.com/api/users/200/:)/607265dbc124241e0f244ef9/kx.png

**Reply by NicolasPLG** [reply]
![4](https://user-images.githubusercontent.com/61419471/114427764-67d88500-9b89-11eb-9cc1-abfa809d76a4.jpg)
![10](https://user-images.githubusercontent.com/61419471/114427767-68711b80-9b89-11eb-82f2-f558adfd81d4.jpg)
![1](https://user-images.githubusercontent.com/61419471/114427768-68711b80-9b89-11eb-9f9d-442371ca08cc.jpg)
![2](https://user-images.githubusercontent.com/61419471/114427769-6909b200-9b89-11eb-93e4-c3a8333ef87a.jpg)
![image](https://user-images.githubusercontent.com/61419471/114427846-7e7edc00-9b89-11eb-939d-e6a3039b02a5.png)
Puedes buscar tambien en la web

@acedward una pregunta la librería **files**, estaría disponible para el web.admin?

**Reply by acedward** [reply]
@NicolasPLG Estamos ahora trabajando en liberar la nueva versión de manejo de archivos. Pronto en what's new!

**Reply by ValeMP** [reply]
@guillevalin me habias mencionado que podía usar "unas que están subidas", te referias a lo que me dice @acedward ??

**Comment by ValeMP** [comment]
@acedward no cache bien si, generando la imagen, hay que poner cierto formato en el body??

**Reply by acedward** [reply]
nada en el body, solo es un GET en la URL puedes cambiar el color / texto / tamaño

**Reply by acedward** [reply]
https://www.cotalker.com/api/users/[size]/[text]/[color-hash]/kx.png

**Reply by ValeMP** [reply]
yaa gracais!!!

**Comment by guillevalin** [comment, marked as answer]
@ValeMP Si entendí bien tu pregunta, no existen imagenes "subidas" como template para asociar a un canal, sin embargo, una solución que hemos usado ampliamente en operaciones es realizar un PATCH al endpoint de /api/channels/:id para setear lo siguiente en el body:

    "image" : {
        "small" : "https://cotalker-us.s3.amazonaws.com/image/1594221568692_small.png", 
        "square" : "https://cotalker-us.s3.amazonaws.com/image/1594221568695_square.png", 
        "original" : "https://cotalker-us.s3.amazonaws.com/image/1594221568696_original.png"
    }

Como "hack" lo que hacíamos era subir una imagen en Cotalker, para obtener el _small, _square y _original, y luego setear esas URLs en los bots.

Otra forma es hacer lo que te propone Eddie, que Cotalker posee URLs dinámicas para generar imagenes.

**Reply by ValeMP** [reply]
yaa gracias!

**Reply by ValeMP** [reply]
@guillevalin para que situación se usa el square?

**Reply by guillevalin** [reply]
@ValeMP Se usa en los avatares de Jitsi (user.avatar.square), y en algunas secciones del admin.

**Reply by ValeMP** [reply]
super gracias
